astropy/units/tests/test_quantity_annotations.py

```diff
@@ -222,3 +222,16 @@ def myfunc_args(solarx: u.arcsec) -> None:
 
     solarx = myfunc_args(1*u.arcsec)
     assert solarx is None
+
+
+def test_enum_annotation():
+    from enum import Enum, auto
```

Comment:
While adding a changelog entry, maybe also add a comment here along the lines of "Regression test for gh-9932".